test/sanity/validate-modules/main.py

```diff
@@ -1097,13 +1132,25 @@ def _validate_argument_spec(self, docs):
                 )
 
             # TODO: needs to recursively traverse suboptions
-            doc_choices = docs.get('options', {}).get(arg, {}).get('choices', [])
-            if not compare_unordered_lists(data.get('choices', []), doc_choices):
+            try:
+                with CaptureStd():
+                    doc_choices = [_type_checker(c) for c in docs.get('options', {}).get(arg, {}).get('choices', [])]
```

Comment:
Would it be worthwhile to report on each choice separately?

Reply:
Stop catching me trying to be lazy ;) 

It probably would be.

Reply:
I've made the change to report on each choice separately